e2e/exhaustive_suite_test.go

```diff
@@ -27,25 +28,54 @@ var Workloads = []workloads.Workload{}
 
 var subscription = &deployers.Subscription{}
 
+var discoveredApps = &deployers.DiscoveredApps{}
+
 // appset := &deployers.ApplicationSet{}
 // Deployers := []deployers.Deployer{subscription, appset}
 
-var Deployers = []deployers.Deployer{subscription}
+var Deployers = []deployers.Deployer{subscription, discoveredApps}
+
+func generateSuffix(storageClassName string) string {
+	suffix := storageClassName
+
+	if strings.ToLower(storageClassName) == "rook-ceph-block" {
+		suffix = "rbd"
+	}
+
+	if strings.ToLower(storageClassName) == "rook-cephfs" {
+		suffix = "cephfs"
+	}
+
+	return suffix
+}
 
 func generateWorkloads([]workloads.Workload) {
 	pvcSpecs := util.GetPVCSpecs()
-	for i, pvcSpec := range pvcSpecs {
+	for _, pvcSpec := range pvcSpecs {
+		// add storageclass name to deployment name
+		suffix := generateSuffix(pvcSpec.StorageClassName)
 		deployment := &workloads.Deployment{
 			Path:     GITPATH,
 			Revision: GITREVISION,
 			AppName:  APPNAME,
-			Name:     fmt.Sprintf("Deployment-%d", i),
+			Name:     fmt.Sprintf("Depl-%s", suffix),
 			PVCSpec:  pvcSpec,
 		}
 		Workloads = append(Workloads, deployment)
 	}
 }
 
+// To skip certain test combination that is not supported
+func skipTest(w workloads.Workload, d deployers.Deployer) bool {
```

Comment:
I would prefer to make use the interface here. Something like this
```
diff --git a/e2e/deployers/deployer.go b/e2e/deployers/deployer.go
index 5b246ebd..87700d44 100644
--- a/e2e/deployers/deployer.go
+++ b/e2e/deployers/deployer.go
@@ -9,6 +9,7 @@ import "github.com/ramendr/ramen/e2e/workloads"
 type Deployer interface {
        Deploy(workloads.Workload) error
        Undeploy(workloads.Workload) error
+       IsWorkloadSupported(workloads.Workload) bool
        // Scale(Workload) for adding/removing PVCs; in Deployer even though scaling is a Workload interface
        // as we can Kustomize the Workload and change the deployer to perform the right action
        // Resize(Workload) for changing PVC(s) size
diff --git a/e2e/deployers/subscription.go b/e2e/deployers/subscription.go
index 916dfcdc..eb38e6ca 100644
--- a/e2e/deployers/subscription.go
+++ b/e2e/deployers/subscription.go
@@ -18,6 +18,10 @@ func (s Subscription) GetName() string {
        return "Subscription"
 }

+func (s Subscription) IsWorkloadSupported(w workloads.Workload) bool {
+       return w.GetName() == "xyz"
+}
+
 func (s Subscription) Deploy(w workloads.Workload) error {
        // Generate a Placement for the Workload
        // Use the global Channel
diff --git a/e2e/exhaustive_suite_test.go b/e2e/exhaustive_suite_test.go
index f280cf8b..453ad684 100644
--- a/e2e/exhaustive_suite_test.go
+++ b/e2e/exhaustive_suite_test.go
@@ -75,6 +75,15 @@ func Exhaustive(t *testing.T) {
 func runTestFlow(t *testing.T) {
        t.Helper()

+       testCtx, err := testcontext.GetTestContext(t.Name())
+       if err != nil {
+               t.Error(err)
+       }
+
+       if !testCtx.Deployer.IsWorkloadSupported(testCtx.Workload) {
+               t.Skip("Workload not supported by deployer")
+       }
+
        if !t.Run("Deploy", DeployAction) {
                t.Fatal("Deploy failed")
        }
```

Reply:
thanks for the suggestion. I updated code based on it.